src/docs/release/breaking-changes/scaffold-messenger.md

```diff
@@ -0,0 +1,182 @@
+---
+title: SnackBars managed by the ScaffoldMessenger
+description: SnackBars are now managed by the ScaffoldMessenger, and persist across routes.
+---
+
+## Summary
+
+The `SnackBar` API within the Scaffold is now handled by the ScaffoldMessenger, one of which is
+available by default within the context of a MaterialApp.
+
+## Context
+
+Prior to this change, `SnackBar`s would be shown by calling on the `Scaffold` within the current
+`BuildContext`. By calling `Scaffold.of(context).showSnackBar`, the current `Scaffold` would animate
+a `SnackBar` into view. This would only apply to the current `Scaffold`, and would not persist across
+routes if they were changed in the course of the `SnackBar`s presentation. This would also lead to
+errors if `showSnackBar` would be called in the course of executing an asynchronous event, and the
+`BuildContext` became invalidated by the route changing and the `Scaffold` being disposed of.
+
+The `ScaffoldMessenger` now handles `SnackBar`s in order to persist across routes and always be
+displayed on the current `Scaffold`. By default, a root `ScaffoldMessenger` is included in the
+`MaterialApp`, but you can create your own controlled scope for the `ScaffoldMessenger` to further
+control _which_ `Scaffold`s receive your `SnackBar`s. 
+
+
+## Description of change
+
+The previous approach called upon the `Scaffold` to show a `SnackBar`.
+
+<!-- skip -->
+```dart
+Scaffold(
+  key: scaffoldKey,
+  body: Builder(
+    builder: (BuildContext context) {
+      return GestureDetector(
+        onTap: () {
+          Scaffold.of(context).showSnackBar(SnackBar(
+            content: const Text('snack'),
+            duration: const Duration(seconds: 1),
+            action: SnackBarAction(
+              label: 'ACTION',
+              onPressed: () { },
+            ),
+          ));
+        },
+        child: const Text('SHOW SNACK'),
```

Comment:
Any particular reason to use all caps here and above? I'm guessing your intent is to make it obvious that the developer can change these, but it just feels shouty... =)

Reply:
Oh I was just copying a sample from the API docs. 😜

Reply:
Fair 😋